code/game/objects/items/devices/miragedrive.dm

```diff
@@ -65,3 +89,66 @@
 	playsound(src.loc, 'sound/weapons/kenetic_reload.ogg', 60, 1)
 	return
 
+/obj/item/mdrive/proc/nyoom(atom/movable/target, list/path, var/lagdist)
+	var/list/testpath = path
+	var/obj/effect/temp_visual/decoy/fading/onesecond/F = new(get_turf(target), target)
+	for(var/i in 1 to length(testpath))
+		var/turf/next_step = testpath[i]
+		if(ISMULTIPLE(i, 2) && (next_step))
+			var/turf/proper_step = testpath[i-lagdist]
+			F.forceMove(proper_step)
+			sleep(0.1 SECONDS)
```

Comment:
sorry, i'm supposed to kill you if you put sleep inside of a loop
you're gonna need to split it into procs and use addtimer